pyomo/repn/tests/cpxlp/test_lpv2.py

```diff
@@ -139,3 +205,180 @@ def test_deterministic_unordered_sets(self):
         self.assertEqual(LOG.getvalue(), "")
 
         self.assertEqual(ref, OUT.getvalue())
+
+    def test_SOS_ordering_bug(self):
+        # This is in response to a bug that was identified with
+        # config.row_order with SOS constraints. If an SOS constraint
+        # was added and the `config.row_order` option selected,
+        # bug would ensue.
+        ref_no_row_order = r"""\* Source Pyomo model name=unknown *\
+
+max 
+x1:
++1 x2
++1 x3
+
+s.t.
+
+c_e_x4_:
++1 x5
+-1 x6
+-2 x7
+-3 x8
+= 0
+
+c_e_x9_:
++1 x10
+-1 x11
+-2 x12
+-3 x13
+= 0
+
+c_e_x14_:
++1 x2
+-1 x6
+-4 x7
+-9 x8
+= 0
+
+c_e_x15_:
++1 x3
+-1 x11
+-4 x12
+-9 x13
+= 0
+
+c_e_x16_:
++1 x6
++1 x7
++1 x8
+= 1
+
+c_e_x17_:
++1 x11
++1 x12
++1 x13
+= 1
+
+c_e_x18_:
++1 x5
+= 2.5
+
+c_e_x19_:
++1 x10
+= 2.0
+
+bounds
+   -inf <= x2 <= +inf
+   -inf <= x3 <= +inf
+   -inf <= x5 <= +inf
+   -inf <= x10 <= +inf
+   0 <= x6 <= +inf
+   0 <= x7 <= +inf
+   0 <= x8 <= +inf
+   0 <= x11 <= +inf
+   0 <= x12 <= +inf
+   0 <= x13 <= +inf
+SOS
+
+x20: S2::
+  x6:1
+  x7:2
+  x8:3
+
+x21: S2::
+  x11:1
+  x12:2
+  x13:3
+
+end
+"""
+        ref_row_order = r"""\* Source Pyomo model name=unknown *\
+
+max 
+x1:
++1 x2
++1 x3
+
+s.t.
+
+c_e_x4_:
++1 x5
++1 x6
++1 x7
+= 1
+
+c_e_x8_:
++1 x9
++1 x10
++1 x11
+= 1
+
+c_e_x12_:
+-1 x5
+-2 x6
+-3 x7
++1 x13
+= 0
+
+c_e_x14_:
+-1 x9
+-2 x10
+-3 x11
++1 x15
+= 0
+
+c_e_x16_:
++1 x2
+-1 x5
+-4 x6
+-9 x7
+= 0
+
+c_e_x17_:
++1 x3
+-1 x9
+-4 x10
+-9 x11
+= 0
+
+c_e_x18_:
++1 x13
+= 2.5
+
+c_e_x19_:
++1 x15
+= 2.0
+
+bounds
+   -inf <= x2 <= +inf
+   -inf <= x3 <= +inf
+   0 <= x5 <= +inf
+   0 <= x6 <= +inf
+   0 <= x7 <= +inf
+   0 <= x9 <= +inf
+   0 <= x10 <= +inf
+   0 <= x11 <= +inf
+   -inf <= x13 <= +inf
+   -inf <= x15 <= +inf
+SOS
+
+x20: S2::
+  x5:1
+  x6:2
+  x7:3
+
+x21: S2::
+  x9:1
+  x10:2
+  x11:3
+
+end
+"""
+        model = create_sos_model()
+        with LoggingIntercept() as OUT1:
+            LPWriter().write(model, OUT1)
+        self.assertEqual(OUT1.getvalue(), ref_no_row_order)
+        with LoggingIntercept() as OUT2:
+            LPWriter().write(model, OUT2, row_order=[model.constraint3])
+        self.assertEqual(OUT2.getvalue(), ref_row_order)
```

Comment:
Please turn on `symbolic_solver_labels` so we can make heads or tails of the baselines.